mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -50,6 +52,9 @@ void LoopNestStateCollector::collect(Operation *opToWalk) {
       loadOpInsts.push_back(op);
     else if (isa<AffineWriteOpInterface>(op))
       storeOpInsts.push_back(op);
+    else if (!mlir::isMemoryEffectFree(op) &&
+             !isa<mlir::memref::AllocOp, mlir::memref::DeallocOp>(op))
+      memoryEffectOpInsts.push_back(op);
```

Comment:
Filtering out alloc and dealloc ops as well. Handling of memref:: alloc/dealloc ops needs to be addressed in a separate issue for now.

Reply:
E.g. Following loop does not merge if %alloc is memref.alloc.
```
  func.func @should_not_fuse_loops_with_side_effects() {
    %alloc = memref.alloc() : memref<10xf32>
    affine.for %arg0 = 0 to 10 {
      %0 = affine.load %alloc[%arg0] : memref<10xf32>
    }
    affine.for %arg0 = 0 to 10 {
      %0 = affine.load %alloc[%arg0] : memref<10xf32>
    }
    return
  }
```